### PR TITLE
chore: export constellation to js

### PR DESCRIPTION
**Pull request by ashneverdawn**
**What this PR does** 📖

Exports constellation to js.
Note `Constellation::put_stream` will be in another PR I'm working on for streaming from js to wasm.

**Which issue(s) this PR fixes** 🔨

Relates to #494

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

File: warp/src/js_exports/constellation.rs [hunk 1]
```diff
@@ -13,3 +16,382 @@ impl ConstellationBox {
...
+    pub fn get_path(&self) -> String {
+        self.inner.get_path().to_str().unwrap().into()
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct Directory {
+    inner: constellation::directory::Directory,
+}
```
Comment (dariusc93) [inline warp/src/js_exports/constellation.rs:104]: Any reason why we couldnt mark `Directory` in `constellation` with `#[wasm_bindgen]` and add the functions there to point to the respective functions?

File: warp/src/js_exports/constellation.rs [hunk 2]
```diff
@@ -13,3 +16,382 @@ impl ConstellationBox {
...
+    pub fn modified(&self) -> js_sys::Date {
+        self.inner.modified().into()
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct File {
+    inner: constellation::file::File,
+}
```
Comment (dariusc93) [inline warp/src/js_exports/constellation.rs:262]: ditto

File: warp/src/js_exports/constellation.rs [hunk 3]
```diff
@@ -13,3 +16,382 @@ impl ConstellationBox {
...
+        self.inner.modified().into()
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone)]
+pub struct Item {
+    file: Option<File>,
+    directory: Option<Directory>,
+}
```
Comment (dariusc93) [inline warp/src/js_exports/constellation.rs:363]: ditto

Comment (dariusc93) [reply warp/src/js_exports/constellation.rs:363]: If we are to use this route, couldnt it have been best to use `Item` in `constellation` instead?

File: warp/src/js_exports/constellation.rs (hunk 2 above)
Comment (ashneverdawn) [reply warp/src/js_exports/constellation.rs:262]: ditto

File: warp/src/js_exports/constellation.rs (hunk 3 above)
Comment (ashneverdawn) [reply warp/src/js_exports/constellation.rs]: Yes, thanks! I've made the change.

**Review by dariusc93 — APPROVED** [review]

**Pull request merged**